### PR TITLE
Added History Manager Section

### PR DESCRIPTION
**Pull request by mtabishk**
1. Added History Manager UI
2. API Integration with backend
 

https://user-images.githubusercontent.com/61789893/192810769-2cb4a4b7-73a0-486f-bb60-accc7bd67d34.mp4

### CODE REVIEW

File: app/lib/screens/medical/manager/history/history_manager.dart [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+      final List<StudentModel> studentsData = [];
+      responseData.forEach((student) {
+        studentsData.add(StudentModel.fromJson(student));
+      });
+
+      if (response.statusCode == 200) {
+        return studentsData;
+      } else {
+        return [];
+      }
```
Comment (ashutoshc8101) [inline app/lib/screens/medical/manager/history/history_manager.dart:44]: The medical manager should be able to search using entry number, not email.

File: app/lib/screens/medical/manager/history/history_manager_detail.dart [hunk 2]
```diff
@@ -0,0 +1,448 @@
...
+        ],
+      ),
+    );
+  }
+}
+
+class DoctorModel {
+  DoctorModel(
+    this.id,
+    this.name,
```
Comment (ashutoshc8101) [inline app/lib/screens/medical/manager/history/history_manager_detail.dart:414]: Move this class to a separate file. One file should only contain one concern.

**Review by ashutoshc8101 — CHANGES_REQUESTED** [review]
Left a few comments. Also, add a video showing the reflected changes in the student side.

File: api/medical/models.py [hunk 3]
```diff
@@ -68,10 +69,6 @@ class MedicalHistory(models.Model):
     patient = models.ForeignKey(
         settings.AUTH_USER_MODEL, on_delete=models.CASCADE)
     doctor = models.ForeignKey(Doctor, on_delete=models.CASCADE)
-    date = models.DateField(null=True)
-    time = models.TimeField(null=True)
-    date = models.DateField(null=True)
-    inhouse = models.BooleanField(default=False)
```
[the remark this answers is not in the document]
Comment (mtabishk) [reply api/medical/models.py:74]: These fields are not required as per UI, I have done these changes with archit.

File: api/medical/serializers.py [hunk 4]
```diff
@@ -28,18 +28,17 @@ class AppointmentSerializer(serializers.Serializer):
     patient = UserSerializer()
     date = serializers.DateField(format='%d %b %Y', input_formats=['%d %b %Y'])
     time = serializers.TimeField(format='%I:%M %p', input_formats='%I:%M %p')
-    start_time = serializers.TimeField(format='%I:%M %p', input_formats='%I:%M %p')
-    end_time = serializers.TimeField(format='%I:%M %p', input_formats='%I:%M %p')
+    start_time = serializers.TimeField(
+        format='%I:%M %p', input_formats='%I:%M %p')
+    end_time = serializers.TimeField(
+        format='%I:%M %p', input_formats='%I:%M %p')
```
[the remark this answers is not in the document]
Comment (mtabishk) [reply api/medical/serializers.py:34]: These changes are by python formatter

File: api/medical/serializers.py [hunk 5]
```diff
@@ -28,18 +28,17 @@ class AppointmentSerializer(serializers.Serializer):
...
     status = serializers.CharField()
     reason = serializers.CharField(max_length=500)
 
 
 class MedicalHistorySerializer(serializers.Serializer):
     patient = UserSerializer()
     doctor = DoctorSerializer()
-    date = serializers.DateField(format='%d %b %Y', input_formats=['%d %b %Y'])
-    time = serializers.TimeField(format='%I:%M %p', input_formats='%I:%M %p')
-    inhouse = serializers.BooleanField()
```
[the remark this answers is not in the document]
Comment (mtabishk) [reply api/medical/serializers.py:42]: Same reason

File: api/medical/views.py [hunk 6]
```diff
@@ -233,13 +233,14 @@ def post(self, request):
...
+            name=name,
+            designation=profession,
+            phone=contact
+        )
+
+        staff.save()
+        return Response(status=200, data={'msg': 'Staff added successfully'})
 
-        staff.save();
-        return Response(status = 200, data={ 'msg': 'Staff added successfully' })
```
[the remark this answers is not in the document]
Comment (mtabishk) [reply api/medical/views.py:242]: These changes are by python formatter

File: api/medical/views.py [hunk 7]
```diff
@@ -366,49 +370,46 @@ def post(self, request):
...
 
         instance = MedicalHistory.objects.create(
             patient=patientinstance,
-            date=date,
             doctor=doctorinstance,
             details=details,
-            inhouse=inhouse,
             diagnosis=diagnosis,
             treatment=treatment,
-            time=time
```
[the remark this answers is not in the document]
Comment (mtabishk) [reply api/medical/views.py:385]: Same reason

File: app/android/app/build.gradle [hunk 8]
```diff
@@ -53,10 +53,10 @@ android {
...
     signingConfigs {
         debug {
-            storeFile file('key.jks')
-            storePassword 'instituteapp'
-            keyAlias 'androiddebugkey'
-            keyPassword 'instituteapp'
+            storeFile file('upload-keystore.jks')
+            storePassword 'password'
+            keyAlias 'uploadkeystore'
+            keyPassword 'password'
```
[the remark this answers is not in the document]
Comment (mtabishk) [reply app/android/app/build.gradle:59]: Okay

File: app/lib/main_dev.dart [hunk 9]
```diff
@@ -21,6 +21,7 @@ Future main() async {
 
   await Firebase.initializeApp(
     options: DefaultFirebaseOptions.currentPlatform,
+    name: 'ira',
```
[the remark this answers is not in the document]
Comment (mtabishk) [reply app/lib/main_dev.dart:24]: App was not running without it on my side. Some duplicate firebase app issue

File: app/lib/screens/medical/manager/history/history_manager.dart (hunk 1 above)
Comment (mtabishk) [reply app/lib/screens/medical/manager/history/history_manager.dart:44]: Medical managers can search using the name. But email is required to fetch list of patients

File: app/lib/screens/medical/manager/history/history_manager_detail.dart (hunk 2 above)
Comment (mtabishk) [reply app/lib/screens/medical/manager/history/history_manager_detail.dart:414]: Okay

**Pull request closed without merging**